third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -392,6 +412,31 @@ struct StoreOpConversion : public ConvertOpToLLVMPattern<triton::StoreOp>,
     const size_t valueElemNBits = dtsize * 8;
 
     const int numVecs = elemsPerThread / vec;
+    // Create cache policy here. Consider moving it to function scope.
+    // createpolicy.fractional.L2::policy.b64 xx 1.0
+    PTXBuilder ptxBuilder;
+    const bool hasL2EvictPolicy =
+        op.getEvict() == triton::EvictionPolicy::EVICT_FIRST ||
+        op.getEvict() == triton::EvictionPolicy::EVICT_LAST;
+    Value policyRet;
+    if (hasL2EvictPolicy) {
+      auto &policy =
+          ptxBuilder.create<>("createpolicy.fractional")
+              ->o("L2::evict_first",
+                  op.getEvict() == triton::EvictionPolicy::EVICT_FIRST)
+              .o("L2::evict_last",
+                 op.getEvict() == triton::EvictionPolicy::EVICT_LAST)
+              .b(64);
+      const std::string writeConstraint = "=l";
+      // prepare asm operands
+      auto *dstOpr = ptxBuilder.newOperand(writeConstraint,
+                                           /*init=*/true);
+      std::string fractionStr = "1.0";
+      auto *fractionOpr = ptxBuilder.newConstantOperand(fractionStr);
+      policy(dstOpr, fractionOpr);
+      Type policyRetTy = rewriter.getI64Type();
+      policyRet = ptxBuilder.launch(rewriter, loc, policyRetTy);
+    }
```

Comment:
looks like the this is the same exact code for load and store. Could you move it in a common helper?

Reply:
Thanks for the quick review! I will add test cases and refactor.

Reply:
are you still working on the refactoring? Let me know when it's ready to review or you can move it into a draft until then